src/dft/backends/mklcpu/commit.cpp

```diff
@@ -25,25 +25,107 @@
+        if (d == domain::COMPLEX) {
+            return DFTI_COMPLEX;
+        }
+        else {
+            return DFTI_REAL;
+        }
+    }
+
+    constexpr DFTI_CONFIG_VALUE get_precision(precision p) {
+        if (p == precision::SINGLE) {
+            return DFTI_SINGLE;
+        }
+        else {
+            return DFTI_DOUBLE;
+        }
+    }
+
+    void set_value(DFTI_DESCRIPTOR_HANDLE& descHandle, dft_values config) {
+        // TODO : add complex storage and workspace, fix error handling
+        status |= DftiSetValue(descHandle, DFTI_INPUT_STRIDES, config.input_strides.data());
+        status |= DftiSetValue(descHandle, DFTI_OUTPUT_STRIDES, config.output_strides.data());
+        status |= DftiSetValue(descHandle, DFTI_BACKWARD_SCALE, config.bwd_scale);
+        status |= DftiSetValue(descHandle, DFTI_FORWARD_SCALE, config.fwd_scale);
+        status |= DftiSetValue(descHandle, DFTI_NUMBER_OF_TRANSFORMS, config.number_of_transforms);
+        status |= DftiSetValue(descHandle, DFTI_INPUT_DISTANCE, config.fwd_dist);
+        status |= DftiSetValue(descHandle, DFTI_OUTPUT_DISTANCE, config.bwd_dist);
+        status |= DftiSetValue(
+            descHandle, DFTI_PLACEMENT,
+            (config.placement == config_value::INPLACE) ? DFTI_INPLACE : DFTI_NOT_INPLACE);
+
+        if (status != DFTI_NO_ERROR) {
+            throw oneapi::mkl::exception("dft", "commit", "DftiSetValue failed");
+        }
+    }
+};
+
+detail::commit_impl* create_commit(descriptor<precision::SINGLE, domain::COMPLEX>& desc) {
+    return new commit_derived_impl<precision::SINGLE, domain::COMPLEX>(desc.get_queue(),
+                                                                       desc.get_values());
 }
-void commit_c(descriptor<precision::SINGLE, domain::COMPLEX> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
+detail::commit_impl* create_commit(descriptor<precision::DOUBLE, domain::COMPLEX>& desc) {
+    return new commit_derived_impl<precision::DOUBLE, domain::REAL>(desc.get_queue(),
+                                                                    desc.get_values());
 }
-void commit_d(descriptor<precision::DOUBLE, domain::REAL> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
+detail::commit_impl* create_commit(descriptor<precision::SINGLE, domain::REAL>& desc) {
+    return new commit_derived_impl<precision::SINGLE, domain::COMPLEX>(desc.get_queue(),
+                                                                       desc.get_values());
 }
-void commit_z(descriptor<precision::DOUBLE, domain::COMPLEX> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
+detail::commit_impl* create_commit(descriptor<precision::DOUBLE, domain::REAL>& desc) {
+    return new commit_derived_impl<precision::DOUBLE, domain::REAL>(desc.get_queue(),
+                                                                    desc.get_values());
```

Comment:
Do we need four identical functions here?
```
template<precision prec, domain dom>
detail::commit_impl* create_commit(descriptor<prec, dom>& desc) {
    return new commit_derived_impl<prec, dom>(desc.get_queue(), desc.get_values());
}

template detail::commit_impl* create_commit<precision::SINGLE, domain::REAL>(descriptor<precision::SINGLE, domain::REAL>&);
template detail::commit_impl* create_commit<precision::SINGLE, domain::COMPLEX>(descriptor<precision::SINGLE, domain::COMPLEX>&);
template detail::commit_impl* create_commit<precision::DOUBLE, domain::REAL>(descriptor<precision::DOUBLE, domain::REAL>&);
template detail::commit_impl* create_commit<precision::DOUBLE, domain::COMPLEX>(descriptor<precision::DOUBLE, domain::COMPLEX>&);
```
or maybe this actually takes up more space. Whatever you think.

Reply:
I think having explicit templatization will reduce compile time

Reply:
I prefer the definition of the function template and explicit instantiations - what Hugh suggested.

Reply:
Explicit instantination of create_commit enabled in aa22374b2296e54167d0a0030e10f60a153dd4a7, along with change in declaration in onemkl_dft_mklxpu.hpp